### ISSUE TITLE
dtype confusion when converting to np.array

### ISSUE THREAD

**Issue by PierreQuinton** [TorchJD/torchjd#235]
Regarding the following line:
https://github.com/TorchJD/torchjd/blob/95bb00d76baa810903f28e7c1bf18906ef631c93/src/torchjd/aggregation/upgrad.py#L116

Here as `gramian_array` comes from torch, it is (at least sometimes) an array with `dtype=float32`.
By default, numpy creates arrays with `dtype=float64`, so `regularization_array` has `dtype=float64`.
The sum of the two has `dtype=float64`.
The solve of the `qp_solver`
https://github.com/TorchJD/torchjd/blob/95bb00d76baa810903f28e7c1bf18906ef631c93/src/torchjd/aggregation/upgrad.py#L131
Seems to fail if the `dtype` of all provided arrays is not `float64`, therefore above if we did not add a regularization array, the solve would fail because the `dtype` is not correct.

While the overall behavior is correct, it made me very confused for a while, I suggest that we cast type of `gramian_array` to `float64`.

**Comment by ValerianRey** [comment]
Yes, we can simply replace `gramian_array = gramian.cpu().detach().numpy()` by `gramian_array = gramian.cpu().detach().numpy().astype(np.float64)` and I think it will do what you want.

Can you check if this problem happens in some other places in the code (some other aggregators mainly)?

And can you make a PR?
